projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,201 @@
+#include "ads1015.h"
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+static void prv_check_status(StatusCode code) {
+  if (!code) {
+    status_msg(code, "failed");
+  }
+}
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(ADS1015Data *data) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  StatusCode code = i2c_write(data->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// writes to register given upper and lower bytes
+static StatusCode prv_setup_register(ADS1015Data *data, uint8_t reg, uint8_t msb, uint8_t lsb) {
+  uint8_t ads1015_setup_register[] = { reg, msb, lsb };
+  StatusCode code = i2c_write(data->i2c_port, data->i2c_addr | ADS1015_I2C_ADDRESS_GND,
+                              ads1015_setup_register, SIZEOF_ARRAY(ads1015_setup_register));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// reads the register and stores the value in the given array(2 bytes)
+static StatusCode prv_read(I2CPort i2c_port, ADS1015Address i2c_addr, uint8_t reg, uint8_t *read,
```

Comment:
Please use more descriptive names - `read` and `size` aren't great names. Something along the lines of `read_buf` and `read_len` or similar would be much more clear. Longer but descriptive variable names are not a bad thing - we have plenty of space in our repo for that! :)